### PR TITLE
fixes #4661 - store hostname with virsh DHCP reservations

### PR DESCRIPTION
**Pull request by domcleal**
@lzap?

### CODE REVIEW

File: lib/proxy/virsh.rb [hunk 1]
```diff
@@ -54,14 +54,21 @@ def virsh_update_dns command, hostname, ip
...
+      xml = "'<host mac=\"#{mac}\" ip=\"#{ip}\"/>'"
+    end
+
     virsh "net-update", net, command, "ip-dhcp-host",
-      "--xml", "'<host mac=\"#{mac}\" ip=\"#{ip}\"/>'",
-      "--live", "--config"
+      "--xml", xml, "--live", "--config"
   rescue Exception => e
-    raise Proxy::DHCP::Error, "Failed to update DNS: #{e}"
+    raise Proxy::DHCP::Error, "Failed to update DHCP: #{e}"
```
Comment (lzap) [inline lib/proxy/virsh.rb:72]: Yeah the problem with this code is it is used both by DNS and DHCP. I guess we can live with this.

Going to test now, thanks.

Comment (domcleal) [inline lib/proxy/virsh.rb:72]: Not in the DHCP method it isn't.. the DNS one is above.

**Comment by lzap** [comment]
Aah okay.

Works great, A+ Thanks

:+1:

**Comment by lzap** [comment]
Merged as 2f75941 for Foreman 1.5. Thanks @domcleal

**Pull request closed without merging**